src/validation/rules/OverlappingFieldsCanBeMergedRule.ts

```diff
@@ -590,17 +592,27 @@ function findConflict(
   const type1 = def1?.type;
   const type2 = def2?.type;
 
-  if (type1 && type2 && doTypesConflict(type1, type2)) {
-    return [
-      [
-        responseName,
-        `they return conflicting types "${inspect(type1)}" and "${inspect(
-          type2,
-        )}"`,
-      ],
-      [node1],
-      [node2],
-    ];
+  if (type1 && type2) {
+    // Errors will have already been handled by RequiredStatusOnFieldMatchesDefinitionRule
+    //  so there's no need to do anything here in the event that modifiedOutputType throws
+    //  an error.
+    try {
+      var modifiedType1 = modifiedOutputType(type1, node1.required);
+      var modifiedType2 = modifiedOutputType(type2, node2.required);
+
+      if (doTypesConflict(modifiedType1, modifiedType2)) {
+        return [
+          [
+            responseName,
+            `they return conflicting types "${inspect(
+              modifiedType1,
+            )}" and "${inspect(modifiedType2)}"`,
+          ],
+          [node1],
+          [node2],
+        ];
+      }
+    } catch {}
```

Comment:
Is it okay not handling errors here if I expect the errors these functions throw to be handled in a different rule? Is there any reason to think one rule will be run and the other won't?